app/models/payments/OpenPaymentsModel.scala

```diff
@@ -18,10 +18,148 @@ package models.payments
 
 import java.time.LocalDate
 
-case class OpenPaymentsModel(paymentType: String,
-                             amount: BigDecimal,
-                             due: LocalDate,
-                             start: LocalDate,
-                             end: LocalDate,
-                             periodKey: String,
-                             overdue: Boolean = false)
+import play.api.i18n.Messages
+import play.api.libs.json.{Json, Writes}
+import views.templates.formatters.dates.DisplayDateRangeHelper.displayDateRange
+
+
+sealed trait OpenPaymentsModel {
+  val chargeType: ChargeType
+  val amount: BigDecimal
+  val due: LocalDate
+  val overdue: Boolean
+  val periodKey: String
+
+  def whatYouOweDescription(implicit messages: Messages): String =  chargeType match {
```

Comment:
Could you add the following above this line to exclude from scalastyle warnings (as I don't imagine there's any more appropriate way of doing this bit of logic):
`//noinspection ScalaStyle`